lms/djangoapps/bulk_enroll/views.py

```diff
@@ -72,9 +83,20 @@ def post(self, request):
                 'action': serializer.data.get('action'),
                 'courses': {}
             }
-            for course in serializer.data.get('courses'):
-                response = students_update_enrollment(self.request, course_id=course)
-                response_dict['courses'][course] = json.loads(response.content)
+            for course_id, cohort_name in itertools.izip_longest(serializer.data.get('courses'),
+                                                                 serializer.data.get('cohorts', [])):
+                response = students_update_enrollment(self.request, course_id=course_id)
+                response_content = json.loads(response.content)
+
+                if cohort_name:
+                    cohort = get_cohort_by_name(course_key=CourseKey.from_string(course_id), name=cohort_name)
+                    for user_data in response_content['results']:
+                        (_, previous_cohort, _) = add_user_to_cohort(cohort, user_data['identifier'])
```

Comment:
All entries in this list may not be valid enrollments. You need to check that `user_data['after']['enrollment'] is True or user_data['after']['allowed'] is True`.

Reply:
agreed!